ecosystem/python/sdk/aptos_sdk/account.py

```diff
@@ -70,10 +77,46 @@ def auth_key(self) -> str:
         """Returns the auth_key for the associated account"""
         return str(AccountAddress.from_key(self.private_key.public_key()))
 
-    def sign(self, data: bytes) -> ed25519.Signature:
+    def sign(self, data: bytes) -> asymmetric_crypto.Signature:
         return self.private_key.sign(data)
 
-    def public_key(self) -> ed25519.PublicKey:
+    def sign_into_simulated_authenticator(self) -> authenticator.Authenticator:
```

Comment:
cc @0xmaayan We should implement something like this for the authenticator classes, that way it's at the class level vs a helper functioncc @0xmaayan We should implement something like this for the authenticator classes, that way it's at the class level vs a helper function